storage/shard_race_prop_test.go

```diff
@@ -132,3 +134,130 @@ func anyIDs() gopter.Gen {
 			return ids
 		})
 }
+
+func TestShardTickWriteRace(t *testing.T) {
+	shard, opts := propTestDatabaseShard(t, 10)
+	defer func() {
+		shard.Close()
+		opts.RuntimeOptionsManager().Close()
+	}()
+
+	ids := []ident.ID{}
+	for i := 0; i < 1; i++ {
+		ids = append(ids, ident.StringID(fmt.Sprintf("foo.%d", i)))
+	}
+
+	var (
+		numRoutines = 1 + /* Fetch */ +1 /* Tick */ + len(ids) /* Write(s) */
+		barrier     = make(chan struct{}, numRoutines)
+		wg          sync.WaitGroup
+	)
+
+	wg.Add(numRoutines)
+
+	for _, id := range ids {
+		id := id
+		go func() {
+			<-barrier
+			ctx := context.NewContext()
+			now := time.Now()
+			require.NoError(t, shard.Write(ctx, id, now, 1.0, xtime.Second, nil))
+			ctx.BlockingClose()
+			wg.Done()
+		}()
+	}
+
+	go func() {
+		<-barrier
+		fetchBlocksMetadataV2ShardFn(shard)
+		wg.Done()
+	}()
+
+	go func() {
+		<-barrier
+		shard.Tick(context.NewNoOpCanncellable())
+		wg.Done()
+	}()
+
+	for i := 0; i < numRoutines; i++ {
+		barrier <- struct{}{}
+	}
+
+	wg.Wait()
+}
+
+func TestShardTickWriteReadRace(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	// seed := time.Now().UnixNano() // int64(1521412723796005124)
```

Comment:
mind un-setting the fixed value and use the time.Now().. version instead - had it set while debugging the particular failure case.

Reply:
Sure thing.